metrics/provider.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	metricsNamespace = "mattermod"
+	httpNamespace    = "requests"
+	cronNamespace    = "cron"
+	githubNamespace  = "github"
+
+	defaultPrometheusTimeoutSeconds = 60
+)
+
+// Provider is the interface that exposes the communication with the metrics system
+// this interface should be implemented by the different providers we want to include
+type Provider interface {
```

Comment:
Shall we rename this into something like `MetricProvider` and move into `server` package? We are losing the interface advatages by coupling this package with server package.

Reply:
Well, the provider is part of the metrics system and is not related only to the server package. Why are we losing interface advantages?